tensorbay/opendataset/AADB/loader.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+
+"""This file defines the AADB Dataloader."""
+
+import os
+from collections import defaultdict
+from typing import Dict
+
+from ...dataset import Data, Dataset
+from ...label import AttributeInfo, Classification
+from ...utility import NameList
+
+_LABEL_FILE_PREFIX = {
+    "new_test": "imgListTestNewRegression",
+    "test": "imgListTestRegression",
+    "train": "imgListTrainRegression",
+    "val": "imgListValidationRegression",
+}
+
+_IMAGE_ROOT_PATH = {
+    "new_test": "AADB_newtest",
+    "test": "datasetImages_warp256",
+    "train": "datasetImages_warp256",
+    "val": "datasetImages_warp256",
+}
+
+_DATASET_NAME = "AADB"
+
+
+def AADB(path: str) -> Dataset:
+    """Load the AADB to TensorBay.
+
+    The file structure looks like:
+    <path>
+        AADB_newtest/
+            0.500_farm1_487_20167490236_ae920475e2_b.jpg
+            ...
+        datasetImages_warp256/
+            farm1_441_19470426814_baae1eb396_b.jpg
+            ...
+        imgListFiles_label/
+            imgList<segment_name>Regression_<attribute_name>.txt
+            ...
+
+    Arguments:
+        path: the root directory of the dataset
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(_DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    classification_attributes_info = dataset.catalog.classification.attributes
+
+    for mode, image_root_path in _IMAGE_ROOT_PATH.items():
+        image_name_handler = (
+            _new_test_image_name_handler if mode == "new_test" else _image_name_handler
+        )
+
+        segment = dataset.create_segment(mode)
+        image_name_attributes_map = _convert_txt_to_dict(
+            root_path, mode, classification_attributes_info
+        )
+        for image_name, attributes in image_name_attributes_map.items():
+            image_path = os.path.join(
+                root_path, image_root_path, image_name_handler(image_name, attributes["score"])
```

Comment:
even if the image_name_handler does not need `score`, the `attributes["score"]` still needs to be called, which is redundant.